File: bthrm.core/Data/Model/HeartRateSession.cs
namespace bthrm.core.Data.Model;

public class HeartRateSession
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public List<HeartRateReading> Readings { get; set; } = new();
}

File: bthrm.web/src/models/HeartRateSession.ts
import type HeartRateReading from "./HeartRateReading";

type HeartRateSession = {
  id: number;
  userId: number;
  startDate: string;
  endDate?: string;
  isPaused: boolean;
  readings: HeartRateReading[];
}

export default HeartRateSession;